### PR TITLE
Model testing

### PR DESCRIPTION
**Pull request by vismate**
I propose merging this branch now, as It is almost complete. The Model.Graph needs tests, but for that to happen I'll need to extend Model.Graph's features, which I plan to do on a separate branch.

### CODE REVIEW

**Review by AMIOL7 — CHANGES_REQUESTED** [review]
I think it would be better to separate tests into separate files, each for single node type, rather than bunching all the tests in one file. This will logically separate tests

**Comment by vismate** [comment]
> I think it would be better to separate tests into separate files, each for single node type, rather than bunching all the tests in one file. This will logically separate tests

Oh, yes I was planning to do that. I don't know why I ended up not including it in this pr.

**Comment by vismate** [comment]
> I think it would be better to separate tests into separate files, each for single node type, rather than bunching all the tests in one file. This will logically separate tests

I ended up not separating each individual node yet. It can be done in the future when each node will have more tests. I think 6-8 files with a couple of tests each would not make too much sense. I created a package for tests regarding the Model and two files: ```ModelTest.java``` which tests connection of multiple nodes, empty inputs / outputs... and ```NodesTests.java``` which contains unit tests for individual nodes.

If you really reckon that each Node's tests should be in a separate file, I can refactor again.

File: app/src/test/java/ELME/ModelTests/NodesTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
+package ELME.ModelTests;
```
Comment (AMIOL7) [inline app/src/test/java/ELME/ModelTests/NodesTest.java]: I would split this file into several files separate for each node type

**Review by AMIOL7 — CHANGES_REQUESTED** [review]
I am happy with the tests themselves

File: app/src/test/java/ELME/ModelTests/NodesTest.java (hunk 1 above)
Comment (vismate) [reply app/src/test/java/ELME/ModelTests/NodesTest.java]: Alrighty, I'll do it then in a few hours.

**Review by AMIOL7 — APPROVED** [review]
Looks good to me

**Review by bgdxqi — APPROVED** [review]

**Comment by bgdxqi** [comment]
Most of the "@author vismate" can be kind of sort of misleading though.

**Comment by vismate** [comment]
> Most of the "@author vismate" can be kind of sort of misleading though.

Yeah, I absolutely get that. I rebased the branch from main, and restructured the project a couple of times, thats why it shows up a lot. I don't know if anything can be done about that.

As far as I'm concerned you are the one who did all of the work on this branch. If there is a way to pin a comment somewhere we should absolutely do that to show who did the real work.

As for the grading... I think the lecturer looks at the commit history which shows everything. And because of how git works no history was ereased by my actions.

**Comment by bgdxqi** [comment]
> > Most of the "@author vismate" can be kind of sort of misleading though.
> 
> Yeah, I absolutely get that. I rebased the branch from main, and restructured the project a couple of times, thats why it shows up a lot. I don't know if anything can be done about that.
> 
> As far as I'm concerned you are the one who did all of the work on this branch. If there is a way to pin a comment somewhere we should absolutely do that to show who did the real work.
> 
> As for the grading... I think the lecturer looks at the commit history which shows everything. And because of how git works no history was ereased by my actions.

Oh, I don't mind it too much. Just wanted to point it out as I was unsure whether you know about it, also I thought you might have some ill-configured settings, which you may have wanted to reconfigure. No biggies at all.

**Pull request merged**